### PR TITLE
Rs/wx/area type wwr

### PR DESCRIPTION
**Pull request by weilixu**
Complete the implementation of area type WWR function with test file (0.0.5)

### CODE REVIEW

**Comment by weilixu** [comment]
@xingcx I made some comments in the RDS for the area type WWR function, could you take a look and confirm the changes?
If needed, we can discuss in the next RDS review meeting

File: rct229/ruleset_functions/get_surface_conditioning_category_dict.py
Comment (charliepnnl) [inline rct229/ruleset_functions/get_surface_conditioning_category_dict.py:4]: Do you want to abbreviate this as you did for SurfaceConditioningCategory?
```
ZoneConditioningCategory as ZCC
```

File: rct229/ruleset_functions/get_surface_conditioning_category_dict.py
Comment (charliepnnl) [inline rct229/ruleset_functions/get_surface_conditioning_category_dict.py:17]: I suggest not including EXTERIOR, GROUND, INTERIOR here because these are not surface conditioning categories.  Ideally, we would get these values from the schema enumerations.  In the mean time, they can be constants in this file outside of the SurfaceConditioningCategory class.

**Review by charliepnnl — APPROVED** [review]
I have made several comments that I hope you will address before merging. But I am approving now to save time.

File: rct229/ruleset_functions/get_area_type_window_wall_area.py
Comment (weilixu) [reply rct229/ruleset_functions/get_area_type_window_wall_area.py:49]: this is in the RDS, I wonder if we can put lower case? I would also prefer lower case

File: rct229/ruleset_functions/get_surface_conditioning_category_dict.py
Comment (weilixu) [reply rct229/ruleset_functions/get_surface_conditioning_category_dict.py:17]: To import the enum, will this looks good?
`SurfaceAdjacentTo = schema_enums["SurfaceAdjacentTo"]`
`                if surface_adjacent_to == SurfaceAdjacentTo.INTERIOR.name
                else surface_adjacent_to,`

File: rct229/ruleset_functions/get_area_type_window_wall_area.py
Comment (weilixu) [reply rct229/ruleset_functions/get_area_type_window_wall_area.py]: Black forced the separate imports.
do it 
`from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
    get_surface_conditioning_category_dict,
    SurfaceConditioningCategory as SCC,
)`
will fail the black format..

Comment (charliepnnl) [reply rct229/ruleset_functions/get_area_type_window_wall_area.py]: Interesting.  I just tried it and it works fine for me.

File: rct229/ruleset_functions/get_surface_conditioning_category_dict.py
Comment (charliepnnl) [reply rct229/ruleset_functions/get_surface_conditioning_category_dict.py]: That works.  At some point, we should try to find a more elegant syntax for handling the enumerations. The `.name` part is just ugly. Probably, instead of using Python's Enumeration type, we should just create a class for each enumeration. Then the syntax could simply be `SurfaceAdjacentTo.INTERIOR` without the `.name`.

I think all we will need to do is to replace line 38 of rct229/data/schema_enums.py with:
```
schema_enums = {
    class_name: type(class_name, (object, ),  enum_dict) 
    for class_name, enum_dict in _enum_dicts.items()}
```
The `type(class_name, (object, ),  enum_dict)` part creates a new type (class) with the given class name, that inherits from `object`, and has class variables given by `enum_dict`.

Let me know if you'd like me to make this change and update wherever schema_enums is used.

**Review by charliepnnl — APPROVED** [review]

File: rct229/ruleset_functions/get_surface_conditioning_category_dict.py
Comment (weilixu) [reply rct229/ruleset_functions/get_surface_conditioning_category_dict.py]: Yea, could you update the method? I agree the .name is ugly and the functioin in schma_enums should takes care of the .name. Thanks!

**Pull request merged**
